src/analyses/base.ml

```diff
@@ -87,14 +87,50 @@ struct
 
   let is_privglob v = GobConfig.get_bool "annotation.int.privglobs" && v.vglob
 
-  let project_val p_opt value is_glob =
-    match GobConfig.get_bool "annotation.int.enabled", is_glob, p_opt with
-    | true, true, _ -> VD.project PU.max_int_precision value
-    | true, false, Some p -> VD.project p value
-    | _ -> value
+  (*This is a bit of a hack to be able to change array domains if a pointer to an array is given as an argument*)
+  (*We have to prevent different domains to be used at the same time for the same array*)
+  (*After a function call, the domain has to be the same as before and we can not depend on the pointers staying the same*)
+  (*-> we determine the arrays a pointer can point to once at the beginning of a function*)
+  (*There surely is a better way, because this means that often the wrong one gets chosen*)
+  module VarMap = Map.Make(CilType.Varinfo)
+  let array_map = ref VarMap.empty
```

Comment:
This is just a `Hashtbl`.

Reply:
There's now this:
```ocaml
  module VarMap = Map.Make(CilType.Varinfo)
  let array_map = Hashtbl.create 20
```
This `Hashtbl` is polymorphic, which isn't suitable for CIL types. Instead it should be something like
```ocaml
  module VarH = Hashtbl.Make(CilType.Varinfo)
  let array_map = VarH.create 20
```

Reply:
> This `Hashtbl` is polymorphic, which isn't suitable for CIL types
I do not understand what is meant here.

To clarify: This is a mapping in two steps: array_map maps a function varinfo to a map from an array varinfo to attributes. Do you mean I should make the second map a Hashtbl as well?

Reply:
Ah, right. I think it's fine if the outer hashmap is such `VarH` and the inner map is that `VarMap`.